src/login/exchangeCodeForToken.ts

```diff
@@ -0,0 +1,56 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Environment } from "@azure/ms-rest-azure-env";
+import { Disposable, EventEmitter, Uri, UriHandler, window } from "vscode";
+import { AuthProviderBase } from "./AuthProviderBase";
+
+class UriEventHandler extends EventEmitter<Uri> implements UriHandler {
+	public handleUri(uri: Uri): void {
+		this.fire(uri);
+	}
+}
+
+const handler: UriEventHandler = new UriEventHandler();
+window.registerUriHandler(handler);
+
+export async function exchangeCodeForToken<TLoginResult>(authProvider: AuthProviderBase<TLoginResult>, clientId: string, environment: Environment, tenantId: string, callbackUri: string, state: string): Promise<TLoginResult> {
```

Comment:
This is not exactly what I meant in terms of refactoring the files, but it seems like the problem is too big to solve in this PR anyways. Fix the uri handler stuff and that should be good for this PR

I don't have a suggested fix, but at a high level my concerns with the file structure are:
- Its hard to know the differences between all the "login" methods
- I don't understand the line between `AuthProviderBase` and `AzureLoginHelper`
- `AzureLoginHelper` is just such a giant mess of random code
- It's not clear which methods deserve their own file and which stay part of a class